src/components/dropdowns/primary-action-dropdown/primary-action-dropdown.js

```diff
@@ -211,89 +208,82 @@ Option.defaultProps = {
   when the dropdown trigger itself is clicked. Otherwise it would open and close
   immediately.
  */
-export class PrimaryActionDropdown extends React.PureComponent {
-  static displayName = 'PrimaryActionDropdown';
-
-  static propTypes = {
-    children: PropTypes.node.isRequired,
-  };
-
-  state = {
-    isOpen: false,
-  };
+const PrimaryActionDropdown = props => {
+  const ref = React.useRef();
+  const [isOpen, setIsOpen] = React.useState(false);
```

Comment:
Use the new collapsible/toggle hook?

Reply:
sure, but need to merge that first 😉

Reply:
Actually we can't, because in the `handleGlobalClick` we force to be `false`, so it's not really a toggle anymore